cpp/examples/tutorial_examples/arrow_example.cc

```diff
@@ -0,0 +1,90 @@
+#include <arrow/api.h>
+#include <arrow/result.h>
+#include <arrow/status.h>
+#include <arrow/table.h>
+
+#include <iostream>
+
+arrow::Status RunMain(int argc, char** argv) {
```

Comment:
Given that we use macros for this in "normal" code, I'm not quite sure how to show them off, in a sense. The biggest difference is generally that you use ARROW_RETURN_NOT_OK for procedures that return Status, and ARROW_ASSIGN_OR_RAISE otherwise. I have methods that do both, so maybe I could just call out when it's happening?

Reply:
Hmm, fair enough. I think calling it out once in the first tutorial and pointing to API docs/cookbook works, then.